src/containers/collaborations/components/DataSetListCard.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+import React from 'react';
+
+import styled from 'styled-components';
+import { faChevronRight } from '@fortawesome/pro-regular-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Map, fromJS, get } from 'immutable';
+import { Button, Colors } from 'lattice-ui-kit';
+import { ValidationUtils, useGoToRoute } from 'lattice-utils';
+import type { UUID } from 'lattice';
+
+import {
+  BasicListCard,
+  DataSetTitle,
+  GapGrid,
+  SpaceBetweenGrid,
+} from '../../../components';
+import { Routes } from '../../../core/router';
+import { ID, ORGANIZATION_ID } from '../../../utils/constants';
+
+const { isValidUUID } = ValidationUtils;
+
+const { NEUTRAL } = Colors;
+
+const ListCard = styled(BasicListCard)`
+  a {
+    color: ${NEUTRAL.N700};
+    text-decoration: none;
+  }
+
+  div > div:last-child > button:first-child {
+    background-color: ${NEUTRAL.N100};
+    margin-right: 10px;
+    padding: 6px 12px;
+
+    :hover {
+      background-color: ${NEUTRAL.N200};
+    }
+  }
+`;
+
+const ContentGrid = styled(SpaceBetweenGrid)`
+  width: 100%;
+`;
+
+const DataSetListCard = ({
+  dataSet,
+  removeDataSet
+} :{
+  dataSet :Map;
+  removeDataSet :(dataSetId :UUID) => void;
+}) => {
+  const dataSetId = get(dataSet, ID);
+  const organizationId = get(dataSet, ORGANIZATION_ID);
+
+  const dataSetRoute = Routes.ORG_DATA_SET
+    .replace(Routes.ORG_ID_PARAM, organizationId)
+    .replace(Routes.DATA_SET_ID_PARAM, dataSetId);
+
+  const goToOrganizationDataSet = useGoToRoute(dataSetRoute);
+
+  const removeDataSetFromCollaboration = (e :SyntheticInputEvent<HTMLInputElement>) => {
+    e.stopPropagation();
+    removeDataSet(dataSet);
+  };
+
+  if (!isValidUUID(dataSetId)) {
+    return null;
+  }
+
+  return (
+    <ListCard key={dataSetId} onClick={goToOrganizationDataSet}>
+      <ContentGrid>
+        <a aria-label="To Data Set Details" href={dataSetRoute}><DataSetTitle dataSet={fromJS(dataSet)} /></a>
```

Comment:
Why isn't this a `react-router` route?

Reply:
I don't think you need to do `fromJS()` here. Also, prop types say it's an immutable map.